🚀 Initial Commit

RUGMJ opened the pull request.

Matthias1590 committed:
Initial commit

RUGMJ committed:
🚀 Initial Commit

RUGMJ committed:
added express typings

RUGMJ committed:
adds module interop to typescript config

RUGMJ committed:
added start script

RUGMJ committed:
makes ts compile as a esmodule instead of CommonJS

RUGMJ committed:
setup basic routes

RUGMJ committed:
adds postgres as a dependency

RUGMJ committed:
setup postgres

RUGMJ committed:
Adds session endpoints

RUGMJ committed:
Adds tokens

RUGMJ committed:
fixes status typo

RUGMJ committed:
combined crash and error into one endpoint

RUGMJ committed:
swapped database packages

RUGMJ committed:
Swaps knex for typeorm

RUGMJ committed:
Sets up typeorm and entities

RUGMJ committed:
Finishes verifyTokenMiddleware

RUGMJ committed:
Adds the session id to the token

Matthias1590 commented on src/entity/Action.ts:
```diff
@@ -0,0 +1,32 @@
+	@ManyToOne(() => Session, session => session.actions)
+	session: Session;
+
+	@OneToMany(() => Command, command => command.action)
```
Why is this a one to many relationship? commands and exceptions should inherit action

Matthias1590 commented on src/entity/Action.ts:
```diff
@@ -0,0 +1,32 @@
+	created_at: Date;
+
+	@Column()
+	updated_at: Date;
```
We don't need this

Matthias1590 commented on src/entity/Action.ts:
```diff
@@ -0,0 +1,32 @@
+	exceptions: Exception[];
+
+	@Column()
+	created_at: Date;
```
Might be nice to rename this to reported_at if we rename the Action model to Event

Matthias1590 commented on src/entity/Action.ts:
```diff
@@ -0,0 +1,32 @@
+} from 'typeorm';
+
+@Entity()
+export class Action {
```
Maybe we should rename this to Event since not all actions (such as exceptions) are actual actions

Matthias1590 commented on src/entity/Command.ts:
```diff
@@ -0,0 +1,20 @@
+	@PrimaryGeneratedColumn()
+	id: number;
+
+	@ManyToOne(() => Action, action => action.commands)
```
Command should just inherit action

Matthias1590 commented on src/entity/Command.ts:
```diff
@@ -0,0 +1,20 @@
+	command: string;
+
+	@Column()
+	created_at: Date;
```
We dont need this

Matthias1590 commented on src/entity/Command.ts:
```diff
@@ -0,0 +1,20 @@
+	created_at: Date;
+
+	@Column()
+	updated_at: Date;
```
We dont need this

Matthias1590 commented on src/entity/Exception.ts:
```diff
@@ -0,0 +1,23 @@
+	id: number;
+
+	@ManyToOne(() => Action, action => action.exceptions)
+	action: Action;
```
Again, exception should inherit Action

Matthias1590 commented on src/entity/Exception.ts:
```diff
@@ -0,0 +1,23 @@
+	action: Action;
+
+	@Column()
+	stacktrace: string;
```
turns out "stack trace" is 2 words, my bad, should be renamed to stack_trace

Matthias1590 commented on src/entity/Exception.ts:
```diff
@@ -0,0 +1,23 @@
+	is_fatal: boolean;
+
+	@Column()
+	created_at: Date;
```
we dont need this

Matthias1590 commented on src/entity/Exception.ts:
```diff
@@ -0,0 +1,23 @@
+	created_at: Date;
+
+	@Column()
+	updated_at: Date;
```
we dont need this

Matthias1590 commented on src/entity/Session.ts:
```diff
@@ -0,0 +1,17 @@
+	hashed_uuid: string;
+
+	@Column()
+	ended_at: Date;
```
should probably be renamed to end_time or end (ended_at doesnt sound right since this date can be in the future)

RUGMJ replied:
How is it going to be in the future?

Matthias1590 replied:
Once a token gets refreshed the end date will be set to now + 5 minutes

RUGMJ replied:
Gotcha makes sense

Matthias1590 commented on src/entity/Session.ts:
```diff
@@ -0,0 +1,17 @@
+
+	@Column()
+	hashed_uuid: string;
+
```
missing a start_time or start field (shouldnt be named started_at if we rename ended_at to end_time or end)

RUGMJ committed:
Adds changes requested in the review

Matthias1590 commented on src/entity/Session.ts:
```diff
@@ -10,8 +10,11 @@ export class Session {
+	end: Date;
+
+	@OneToMany(() => Event, action => action.session)
+	actions: Event[];
```
should probably be named events

RUGMJ replied:
You're not meant to use plural versions of names in databases

Matthias1590 replied:
I meant the field "actions"

RUGMJ replied:
Oh right, that probably got missed when I mass renamed the value

Matthias1590 commented on src/index.ts:
```diff
@@ -46,9 +46,19 @@ v1.post('/session/create', async (req, res) => {
-	const token = jwt.sign({ hashedServerId }, JWT_SECRET, {
-		expiresIn: EXPIRE_TIME_SECS,
-	});
+	const session = new Session();
```
does the start field get set automatically? and can we maybe require the hashed_uuid in the constructor so we can ensure its set?

RUGMJ replied:
It should but sure

Matthias1590 replied:
If the database throws some exception if the hashed_uuid member isnt set then that works too, just want to make sure it doesnt cause any bugs

Matthias1590 commented on src/index.ts:
```diff
@@ -65,24 +75,20 @@ v1.post('/command', verifyTokenMiddleware, (req, res) => {
-	// });
 
 	const decoded = verifyToken(token);
 	if (decoded === 'NoToken')
```
This should probably just check if status does not indicate success and if it doesnt switch on the response (decoded) instead of multiple ifs

RUGMJ replied:
Yeah, I thought that too after I changed it and forgot before I pushed

RUGMJ committed:
Fits to the requests in the review

RUGMJ committed:
swapped over to crypto instead of bcrypt

Bcrypt didint make fixed hashes
which wouldve caused issues when querying the database

RUGMJ committed:
removes the session id from the tokens

so the client doesnt know how many sessions there are

RUGMJ committed:
implements session/refresh

RUGMJ committed:
Adds the session to req.body

RUGMJ committed:
Implements /exception and /command

RUGMJ committed:
Allows for the port to be configured

RUGMJ committed:
Adds a README.md

RUGMJ committed:
Adds checks for mojang auth on /session/refresh

Matthias1590 commented on src/index.ts:
```diff
@@ -44,16 +44,61 @@ v1.post('/session/create', async (req, res) => {
+		sessionRepository,
+	);
+
+	session.end = new Date(session.end.getTime() + EXPIRE_TIME_SECS * 1000);
```
This shouldnt add to the end but rather to the current time (Date.now())

RUGMJ replied:
Initially I thought the same  but wouldn't that mean that the end gets changed because of delays?

Matthias1590 replied:
Wdym gets changed? Its supposed to get changed, if you add 5 minutes everytime someone makes a request to the refresh endpoint people can just make requests every second and end up with a token that doesnt expire in hours

RUGMJ replied:
True I didnt even think about that

RUGMJ replied:
Although the expiration time on the token is different to the session end value but it probably should still be changed

Matthias1590 replied:
It shouldnt be different, if we return a token with an expiry time of 5 minutes then that has the same expiry time as the session

RUGMJ committed:
fixes the time addition on the end field

RUGMJ commented:
I did not mean to close this
